db/migrate/20220906152545_add_unique_index_participant_declaration.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddUniqueIndexParticipantDeclaration < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def change
+    add_index :participant_declarations, %i[cpd_lead_provider_id participant_profile_id declaration_type course_identifier state], unique: true, name: :unique_declaration_index, algorithm: :concurrently, where: "state NOT IN ('voided', 'ineligible')"
```

Comment:
would it not be better to whitelist rather than blacklist on the `where`?

Reply:
although not likely is is technically possible to clawback multiple times there having multiple `clawed_back` should be permitted

Reply:
👍 sounds sensible